stan/math/prim/scal/fun/grad_reg_inc_gamma.hpp

```diff
@@ -20,20 +21,46 @@ namespace stan {
       using std::fabs;
       using std::log;
 
-      T S = 0;
-      T s = 1;
       T l = log(z);
-      int k = 0;
-      T delta = s / (a * a);
-      while (fabs(delta) > precision) {
-        S += delta;
-        ++k;
-        s *= - z / k;
-        delta = s / ((k + a) * (k + a));
-        if (isinf(delta))
-          throw domain_error("gradRegIncGamma not converging");
+      if (z >= a && z >= (T)8) {
```

Comment:
For the second part of the condition, don't cast to a `T`. That results in an overpromotion. We've implemented comparison operators for double and the Stan autodiff types.

So... change the line to:
`if (z >= a && z >= 8) {`